### ISSUE TITLE
Need help with fixing network configuration, DNS does not work

### ISSUE THREAD

**Issue by FranoseQ** [kevoreilly/CAPEv2#1533]
Need help with fixing network configuration, DNS does not work

I'm having problem with DNS on Guest machine, i can ping any ip perfectly however pinging domains doesnt work .
Can someone please help me fix this issue? I have been stuck here and I'm out of ideas what to do, any help would be amazing [❤️](https://emojipedia.org/red-heart/)

Environment:
Guest VM: Win10 - KVM hypervisor
Host VM: Ubuntu 22.04.2 LTS

Cuckoo log:
```
2023-05-22 09:37:02,450 [lib.cuckoo.core.scheduler] INFO: Task #42: Starting analysis of FILE '/tmp/cuckoo-tmp/upload_qgdycx1k/test-cape-ping.bat'
2023-05-22 09:37:02,453 [lib.cuckoo.core.scheduler] INFO: Task #42: File already exists at '/opt/CAPEv2/storage/binaries/7afb1d17a68698ab3519a59f4c60f00d776409f3106555d5a2274895100c7e26'
2023-05-22 09:37:02,465 [lib.cuckoo.core.scheduler] INFO: Task #42: acquired machine cuckoo1 (label=cuckoo1, arch=x64, platform=windows)
2023-05-22 09:37:02,476 [lib.cuckoo.common.integrations.parse_pe] ERROR: PE type not recognised: 'Unable to read the DOS Header, possibly a truncated file.'
2023-05-22 09:37:14,076 [lib.cuckoo.core.scheduler] INFO: Enabled route 'internet'. Bear in mind that routes none and drop won't generate PCAP file
2023-05-22 09:37:14,085 [modules.auxiliary.sniffer] INFO: Started sniffer with PID 5684 (interface=virbr2, host=192.168.1.158, dump path=/opt/CAPEv2/storage/analyses/42/dump.pcap)
2023-05-22 09:37:14,099 [lib.cuckoo.core.guest] INFO: Task #42: Starting analysis on guest (id=cuckoo1, ip=192.168.1.158)
2023-05-22 09:37:14,455 [lib.cuckoo.core.guest] INFO: Task #42: Guest is running CAPE Agent 0.11 (id=cuckoo1, ip=192.168.1.158)
2023-05-22 09:37:23,253 [lib.cuckoo.core.guest] INFO: Task #42: Uploading support files to guest (id=cuckoo1, ip=192.168.1.158)
2023-05-22 09:37:23,254 [lib.cuckoo.core.guest] INFO: Task #42: Uploading script files to guest (id=cuckoo1, ip=192.168.1.158)
2023-05-22 09:41:10,636 [lib.cuckoo.core.guest] INFO: Task #42: Analysis completed successfully (id=cuckoo1, ip=192.168.1.158)
2023-05-22 09:41:11,703 [lib.cuckoo.core.scheduler] INFO: Disabled route 'internet'
2023-05-22 09:41:11,725 [lib.cuckoo.core.scheduler] INFO: Task #42: analysis procedure completed
```

More information:
-I am able to ping host from guest and guest from host.
-Guest VM - Firewall OFF, Virus&Threat detection OFF, machine prepared as in the documentation
-Agent installed via task scheduler
-I am able to run analysis with network routing "Drop", everything works ok
-For now i configured **internet = ens33** in routing.conf , im not planing on running malware for now, just want to fix network on Guest

Config files:

ifconfig - host
```
virbr2: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 192.168.1.1  netmask 255.255.255.0  broadcast 192.168.1.255
        ether 52:54:00:18:b0:7e  txqueuelen 1000  (Ethernet)
        RX packets 5728  bytes 4500955 (4.5 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5846  bytes 27193557 (27.1 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```
```
ens33: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.46  netmask 255.255.255.0  broadcast 192.168.0.255
        ether 00:0c:29:1b:46:91  txqueuelen 1000  (Ethernet)
        RX packets 1726  bytes 668303 (668.3 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1688  bytes 262402 (262.4 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

cuckoo.conf

```
[cuckoo]

# Which category of tasks do you want to analyze?
categories = static, pcap, url, file

# If turned on, Cuckoo will delete the original file after its analysis
# has been completed.
delete_original = off

# Archives are not deleted by default, as it extracts and "original file" become extracted file
delete_archive = on

# If turned on, Cuckoo will delete the copy of the original file in the
# local binaries repository after the analysis has finished. (On *nix this
# will also invalidate the file called "binary" in each analysis directory,
# as this is a symlink.)
delete_bin_copy = off

# Specify the name of the machinery module to use, this module will
# define the interaction between Cuckoo and your virtualization software
# of choice.
machinery = kvm

# Enable creation of memory dump of the analysis machine before shutting
# down. Even if turned off, this functionality can also be enabled at
# submission. Currently available for: VirtualBox and libvirt modules (KVM).
memory_dump = off

# When the timeout of an analysis is hit, the VM is just killed by default.
# For some long-running setups it might be interesting to terminate the
# moinitored processes before killing the VM so that connections are closed.
terminate_processes = off

# Enable automatically re-schedule of "broken" tasks each startup.
# Each task found in status "processing" is re-queued for analysis.
reschedule = off

# Fail "unserviceable" tasks as they are queued.
# Any task found that will never be analyzed based on the available analysis machines
# will have its status set to "failed".
fail_unserviceable = on

# Limit the amount of analysis jobs a Cuckoo process goes through.
# This can be used together with a watchdog to mitigate risk of memory leaks.
max_analysis_count = 0

# Limit the number of concurrently executing analysis machines.
# This may be useful on systems with limited resources.
# Set to 0 to disable any limits.
max_machines_count = 10

# Limit the amount of VMs that are allowed to start in parallel. Generally
# speaking starting the VMs is one of the more CPU intensive parts of the
# actual analysis. This option tries to avoid maxing out the CPU completely.
max_vmstartup_count = 5

# Minimum amount of free space (in MB) available before starting a new task.
# This tries to avoid failing an analysis because the reports can't be written
# due out-of-diskspace errors. Setting this value to 0 disables the check.
# (Note: this feature is currently not supported under Windows.)
freespace = 50000
# Process tasks, but not reach out of memory
freespace_processing = 15000

# Temporary directory containing the files uploaded through Cuckoo interfaces
# (web.py, api.py, Django web interface).
tmppath = /tmp

# Delta in days from current time to set the guest clocks to for file analyses
# A negative value sets the clock back, a positive value sets it forward.
# The default of 0 disables this option
# Note that this can still be overridden by the per-analysis clock setting
# and it is not performed by default for URL analysis as it will generally
# result in SSL errors
daydelta = 0

# Path to the unix socket for running root commands.
rooter = /tmp/cuckoo-rooter

# Enable if you want to see a DEBUG log periodically containing backlog of pending tasks, locked vs unlocked machines.
# NOTE: Enabling this feature adds 4 database calls every 10 seconds.
periodic_log = off

# Max filename length for submissions, before truncation. 196 is arbitrary.
max_len = 196

# If it is greater than this, call truncate the filename further for sanitizing purposes.
# Length truncated to is controlled by sanitize_to_len.
#
# This is to prevent long filenames such as files named by hash.
sanitize_len = 32
sanitize_to_len = 24

[resultserver]
# The Result Server is used to receive in real time the behavioral logs
# produced by the analyzer.
# Specify the IP address of the host. The analysis machines should be able
# to contact the host through such address, so make sure it's valid.
# NOTE: if you set resultserver IP to 0.0.0.0 you have to set the option
# `resultserver_ip` for all your virtual machines in machinery configuration.
ip = 192.168.1.1

# Specify a port number to bind the result server on.
port = 2042

# Force the port chosen above, don't try another one (we can select another
# port dynamically if we can not bind this one, but that is not an option
# in some setups)
force_port = yes

pool_size = 0

# Should the server write the legacy CSV format?
# (if you have any custom processing on those, switch this on)
store_csvs = off

# Maximum size of uploaded files from VM (screenshots, dropped files, log)
# The value is expressed in bytes, by default 100MB.
upload_max_size = 100000000

# To enable trimming of huge binaries go to -> web.conf -> general -> enable_trim
# Prevent upload of files that passes upload_max_size?
do_upload_max_size = no

[processing]
# Set the maximum size of analyses generated files to process. This is used
# to avoid the processing of big files which may take a lot of processing
# time. The value is expressed in bytes, by default 200MB.
analysis_size_limit = 200000000

# The number of calls per process to process. 0 switches the limit off.
# 10000 api calls should be processed in less than 2 minutes
analysis_call_limit = 0

# Enable or disable DNS lookups.
resolve_dns = on

# Enable or disable reverse DNS lookups
# This information currently is not displayed in the web interface
reverse_dns = off

# Use ram to boost processing speed. You will need more than 20GB of RAM for this feature.
# Please read "performance" section in the documentation.
ram_boost = no

# Enable PCAP sorting, needed for the connection content view in the web interface.
sort_pcap = on

[database]
# Specify the database connection string.
# Examples, see documentation for more:
# sqlite:///foo.db
# postgresql://foo:bar@localhost:5432/mydatabase
# mysql://foo:bar@localhost/mydatabase
# If empty, default is a SQLite in db/cuckoo.db.
# SQLite doens't support database upgrades!
# For production we strongly suggest go with PostgreSQL
connection = postgresql://cape:SuperPuperSecret@localhost:5432/cape

# Database connection timeout in seconds.
# If empty, default is set to 60 seconds.
timeout =

[timeouts]
# Set the default analysis timeout expressed in seconds. This value will be
# used to define after how many seconds the analysis will terminate unless
# otherwise specified at submission.
default = 200

# Set the critical timeout expressed in (relative!) seconds. It will be added
# to the default timeout above and after this timeout is hit
# Cuckoo will consider the analysis failed and it will shutdown the machine
# no matter what. When this happens the analysis results will most likely
# be lost.
critical = 60

# Maximum time to wait for virtual machine status change. For example when
# shutting down a vm. Default is 300 seconds.
vm_state = 300

[log_rotation]
# Activate log rotation for cuckoo.log and process.log.
enabled = on
# Keep 30 days of log history (default is 7).
backup_count = 30

[tmpfs]
# only if you using volatility to speedup IO
# mkdir -p /mnt/tmpfs
# mount -t tmpfs -o size=50g ramfs /mnt/tmpfs
# chown cape:cape /mnt/tmpfs
#
# vim /etc/fstab
# tmpfs       /mnt/tmpfs tmpfs   nodev,nosuid,noexec,nodiratime,size=50g   0 0
#
# Add crontab with
# @reboot chown cape:cape /mnt/tmpfs -R
enabled = off
path = /mnt/tmpfs/
# in mb
freespace = 2000
```
kvm.conf
```
[kvm]
# Specify a comma-separated list of available machines to be used. For each
# specified ID you have to define a dedicated section containing the details
# on the respective machine. (E.g. cuckoo1,cuckoo2,cuckoo3)
machines = cuckoo1,ubuntu_x64

interface = virbr2

[ubuntu_x64]

label = ubuntu_x64
platform = linux
ip = 192.168.100.2
arch = x64 
snapshot = ubuntu64-snap8

[cuckoo1]
# Specify the label name of the current machine as specified in your
# libvirt configuration.
label = cuckoo1

# Specify the operating system platform used by current machine
# [windows/darwin/linux].
platform = windows

# Specify the IP address of the current virtual machine. Make sure that the
# IP address is valid and that the host machine is able to reach it. If not,
# the analysis will fail. You may want to configure your network settings in
# /etc/libvirt/<hypervisor>/networks/
ip = 192.168.1.158

# (Optional) Specify tags to display
# Tags may be used to specify on which guest machines a sample should be run
# tags = windows_xp_sp3,acrobat_reader_6

# (Optional) Specify the snapshot name to use. If you do not specify a snapshot
# name, the KVM MachineManager will use the current snapshot.
# Example (Snapshot1 is the snapshot name):
snapshot = snapshot3

# (Optional) Specify the name of the network interface that should be used
# when dumping network traffic from this machine with tcpdump. If specified,
# overrides the default interface specified in auxiliary.conf
# Example (virbr0 is the interface name):
interface = virbr2

# (Optional) Specify the IP of the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here
# the IP address for the Result Server as your machine sees it. If you don't specify an
# address here, the machine will use the default value from cuckoo.conf.
# NOTE: if you set this option you have to set result server IP to 0.0.0.0 in cuckoo.conf.
# Example:
resultserver_ip = 192.168.1.1

# (Optional) Specify the port for the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here
# the port for the Result Server as your machine sees it. If you don't specify a port
# here, the machine will use the default value from cuckoo.conf.
# Example:
resultserver_port = 2042

# Set the machine architecture
# Required to auto select proper machine architecture for sample
# x64 or x86
arch = x64

# (Optional) Specify whether or not the machine should be reserved, meaning that it will
# only be used for a detonation if specifically requested by its label.
# reserved = no

```
routing.conf 
```
[routing]

# Enable pcap generation for non live connections?
# If you have huge number of VMs, pcap generation can be a bottleneck
enable_pcap = no

# Default network routing mode; "none", "internet", or "vpn_name".
# In none mode we don't do any special routing - the VM doesn't have any
# network access (this has been the default actually for quite a while).
# In internet mode by default all the VMs will be routed through the network
# interface configured below (the "dirty line").
# And in VPN mode by default the VMs will be routed through the VPN identified
# by the given name of the VPN.
# Note that just like enabling VPN configuration setting this option to
# anything other than "none" requires one to run utils/rooter.py as root next
# to the CAPE instance (as it's required for setting up the routing).
route = internet

# Network interface that allows a VM to connect to the entire internet, the
# "dirty line" so to say. Note that, just like with the VPNs, this will allow
# malicious traffic through your network. So think twice before enabling it.
# (For example, to route all VMs through eth0 by default: "internet = eth0").
internet = ens33

# Routing table name/id for "dirty line" interface. If "dirty line" is
# also default gateway in the system you can leave "main" value. Otherwise add
# new routing table by adding "<id> <name>" line to /etc/iproute2/rt_tables
# (e.g., "200 eth0"). ID and name must be unique across the system (refer to
# /etc/iproute2/rt_tables for existing names and IDs).
rt_table = main

# When using "dirty line", you can reject forwarding to a certain network segment.
# For example, a request targeting 192.168.12.1/24,172.16.22.1/24 will not be
# forwarded, but will be rejected:
# "reject_segments = 192.168.12.1/24,172.16.22.1/24"
reject_segments = none

# When ussing "dirty line", you can reject guest access a certain port.
# For example, a request targeting host's port 8000 and 8080 will be rejected:
# "reject_hostports = 8000,8080"
reject_hostports = none

# To route traffic through multiple network interfaces CAPE uses
# Policy Routing with separate routing table for each output interface
# (VPN or "dirty line"). If this option is enabled CAPE on start will try
# to automatically initialise routing tables by copying routing entries from
# main routing table to the new routing tables. Depending on your network/vpn
# configuration this might not be sufficient. In such case you would need to
# initialise routing tables manually. Note that enabling this option won't
# affect main routing table.
auto_rt = no

# The drop route basically drops any outgoing network (except for CAPE
# traffic) whereas the regular none route still allows a VM to access its own
# subnet (e.g., 192.168.122.1/24). It is disabled by default as it does require
# the optional rooter to run (unlike the none route, where literally nothing
# happens). One can either explicitly enable the drop route or if the rooter
# is enabled anyway, it is automatically enabled.
drop = no

# Should check if the inteface is up
verify_interface = yes

[inetsim]
# Inetsim quick deploy, chose your vm manager if is not kvm
# wget https://googledrive.com/host/0B6fULLT_NpxMQ1Rrb1drdW42SkE/remnux-6.0-ova-public.ova
# tar xvf remnux-6.0-ova-public.ova
# qemu-img convert -O qcow2 REMnuxV6-disk1.vmdk remnux.qcow2

enabled = no
server = 192.168.1.2
dnsport = 53
interface = virbr1
# Redirect TCP ports (should we also support UDP?). If specified, this should
# represent whitespace-separated src:dst pairs. E.g., "80:8080 443:8080" will
# redirect all 80/443 traffic to 8080 on the specified InetSim host.
# Source port range redirection is also supported. E.g., "996-2041:80" will
# redirect all traffic directed at ports between 996 and 2041 inclusive to port 80
# on the specified InetSim host.
ports =


[tor]
enabled = no
dnsport = 5353
proxyport = 9040
interface = virbr1

[vpn]
# By default we disable VPN support as it requires running utils/rooter.py as
# root next to cuckoo.py (which should run as regular user).
enabled = no

# select one of the configured vpns randomly
random_vpn = no

# Comma-separated list of the available VPNs.
vpns = vpn0

[vpn0]
# Name of this VPN. The name is represented by the filepath to the
# configuration file, e.g., cuckoo would represent /etc/openvpn/cuckoo.conf
# Note that you can't assign the names "none" and "internet" as those would
# conflict with the routing section in cuckoo.conf.
name = vpn0

# The description of this VPN which will be displayed in the web interface.
# Can be used to for example describe the country where this VPN ends up.
description = openvpn_tunnel

# The tun device hardcoded for this VPN. Each VPN *must* be configured to use
# a hardcoded/persistent tun device by explicitly adding the line "dev tunX"
# to its configuration (e.g., /etc/openvpn/vpn1.conf) where X in tunX is a
# unique number between 0 and your lucky number of choice.
interface = tun0

# Routing table name/id for this VPN. If table name is used it *must* be
# added to /etc/iproute2/rt_tables as "<id> <name>" line (e.g., "201 tun0").
# ID and name must be unique across the system (refer /etc/iproute2/rt_tables
# for existing names and IDs).
rt_table = tun0


[socks5]
# By default we disable socks5 support as it requires running utils/rooter.py as
# root next to cuckoo.py (which should run as regular user).
enabled = no

# select one of the configured socks5 proxies randomly
random_socks5 = no

# Comma-separated list of the available proxies.
proxies = socks_ch

[socks_ch]
name = ch_socks
description = ch_socks
proxyport = 5008
dnsport = 10053
```

Network in virt-manager
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/99bd2f13-6e61-4121-890d-7d42c4bdec69)

Network in VM settings
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/82360bef-be33-47a1-b7a7-ebc36fc88c4d)

ifconfig Guest VM:
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/14d209f7-e30e-4d4c-b55e-e1943b635408)

Network settings Guest VM
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/78ae026c-4dc2-48ae-a39d-6c88314207d1)

systemctl status dnsmasq
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/f6377e18-460c-4729-b61d-fe62b9317342)

systemctl status  cape-rooter.service
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/8ec1fc55-8484-43c9-b9dd-3aa4a150ddae)

Test ping form GuestVM
![image](https://github.com/kevoreilly/CAPEv2/assets/100945246/b1be0e0b-c5fa-484e-b237-51a1bb39dc75)

**Comment by doomedraven** [comment]
First of all you madre me happy, didnt see in long time any proper issue filled with useful info.

I wont set dns to cape box if you dont want to configure your own dns forwarder, just set it to public dns inside of the  vm

**Comment by FranoseQ** [comment]
Ok, i get it now, everything works fine on google dns. Thank you :)
